src/Conductor/Controllers/DefinitionController.cs

```diff
@@ -24,9 +24,21 @@ public DefinitionController(IDefinitionService service)
 
         [HttpGet]
         [Authorize(Policy = Policies.Author)]
-        public ActionResult<IEnumerable<string>> Get()
+        public ActionResult<IEnumerable<Definition>> Get([FromQuery] PaginationParameter parameter)
         {
-            return new string[] { "value1", "value2" };
+            var result = parameter.IsValid() ?
+
+                _service.GetDefinitions()
+                    .OrderByDescending(x => x.Version)
+                    .DistinctBy(p => p.Id)
+                    .Skip((parameter.PageNumber - 1) * parameter.PageSize)
+                    .Take(parameter.PageSize) :
+
+                _service.GetDefinitions()
+                    .OrderByDescending(x => x.Version)
+                    .DistinctBy(p => p.Id);       
```

Comment:
This will cause the app to pull everything from the DB over the network and to the query in memory.

Reply:
True, I got tricked by the _collection.AsQueryable() from repository, but now I see that we are not returning an IQueriable() at all.